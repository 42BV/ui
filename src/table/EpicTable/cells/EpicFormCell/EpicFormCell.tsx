import React from 'react';

interface Props {
  /**
   * The content of the cell.
   */
  children: React.ReactNode;

  /**
   * The width of the cell.
   */
  width: number;

  /**
   * The height of the cell.
   */
  height: number;
}

/**
 * The EpicFormCell is used inside of a EpicForm to render form inputs in.
 * It can be seen as the EpicForm's variant of the `<td>` element.
 */
export function EpicFormCell({ children, width, height }: Props) {
  return (
    <div
      className={'epic-table-form-cell border-bottom py-1 pr-1'}
      style={{
        minWidth: width,
        width,
        height,
        paddingLeft: '0px'
      }}
    >
      {children}
    </div>
  );
}
