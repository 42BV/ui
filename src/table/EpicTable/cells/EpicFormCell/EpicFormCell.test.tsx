import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { EpicFormCell } from './EpicFormCell';

describe('Component: EpicFormCell', () => {
  test('ui', () => {
    const epicFormCell = shallow(
      <EpicFormCell width={200} height={48}>
        just a test
      </EpicFormCell>
    );
    expect(toJson(epicFormCell)).toMatchSnapshot();
  });
});
