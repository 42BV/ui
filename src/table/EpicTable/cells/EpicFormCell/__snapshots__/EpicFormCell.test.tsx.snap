// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EpicFormCell ui 1`] = `
<div
  className="epic-table-form-cell border-bottom py-1 pr-1"
  style={
    Object {
      "height": 48,
      "minWidth": 200,
      "paddingLeft": "0px",
      "width": 200,
    }
  }
>
  just a test
</div>
`;
